src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java

```diff
@@ -500,6 +508,16 @@ private boolean validCsra(final Assessment firstAssessment) {
         return (firstAssessment.isCellSharingAlertFlag() && !"PEND".equals(firstAssessment.getClassificationCode()));
     }
 
+    private boolean isReviewedCsra(final AssessmentDto assessmentDto) {
+        return (assessmentDto.getReviewSupLevelType() != null && !"PEND".equals(assessmentDto.getReviewSupLevelTypeDesc()));
+    }
+
+    private boolean isCalculatedCsra(final AssessmentDto assessmentDto) {
+        return (assessmentDto.getCalcSupLevelType() != null && !"PEND".equals(assessmentDto.getCalcSupLevelTypeDesc()))
+            && assessmentDto.getOverridedSupLevelType() == null;
+    }
+
+
```

Comment:
I suppose if this Kotlin we would have added some extension functions to the AssessmentsDTO to improve readability with the not/and/negation above 

`fun AssessementDto.hasNonPendingReview() ` or whatever